### PR TITLE
Added import and export function

### PR DESCRIPTION
**Pull request by Ponkhy**
I added a simple import and export function to backup the monitors and notifications.
I didn't include the assignments which are on the monitors, i think it would be too much of a hassle because if the database keys.
Of course it would also be possible to just import everything directly through sql inserts but i think this is the cleaner way.

Also for some reason i didn't find out why the hover on the `type="file"` doesn't work, maybe someone else have a idea.
What do you guys think?

**New i18n keys:**
`Import/Export Backup`
`Export`
`Import`
`backupDescription`
`backupDescription2`
`backupDescription3`
`alertNoFile`
`alertWrongFileType`

### CODE REVIEW

**Comment by louislam** [comment]
> Also for some reason i didn't find out why the hover on the type="file" doesn't work, maybe someone else have a idea.
What do you guys think?

Fixed. Have to override bootstrap's selector. Global search found this.

**Review by Saibamen — CHANGES_REQUESTED** [review]

File: src/pages/Settings.vue [hunk 1]
```diff
@@ -293,6 +301,41 @@ export default {
             this.$root.storage().removeItem("token");
         },
 
+        downloadBackup() {
+            let time = dayjs().format("YYYY_MM_DD-hh_mm_ss");
+            let fileName = `Uptime_Kuma_Backup_${time}.json`;
+            let exportData = {
+                notificationList: this.$root.notificationList,
+                monitorList: this.$root.monitorList,
+            }
```
[the remark this answers is not in the document]
Comment (Ponkhy) [reply src/pages/Settings.vue:310]: Do you mean the Uptime Kuma Version, or a own one for backups?

[the remark this answers is not in the document]
Comment (Saibamen) [reply src/pages/Settings.vue:310]: IMO just Kuma version for now

**Review by Saibamen — APPROVED** [review]

**Comment by Ponkhy** [comment]
Should be done now, can't think of any further changes.

**Comment by gaby** [comment]
Really looking forward to test this feature! We have hundreds of hosts to monitor, and having the ability to import would help a lot.

**Comment by louislam** [comment]
1. Paused monitors are not pause after restore, it is expected?
2. Error handling for the "Import" button:

- Not select file
- Selected a mp4 file
- Selected a txt file with "Hello world"
- Selected a txt file with "{}" 

3. Might need to show "Importing"

Thanks!

**Comment by Ponkhy** [comment]
Alright, should be fixed now!

While importing you'll see a spinner and the buttom will be disabled.
I also added error massages when you try to import a non-json file or just didn't select one.
And paused monitors will stay paused after they got imported.

**Comment by louislam** [comment]
Found another issue, it seems that all spaces in any string are lost after imported.

For example: "Test Webhook" => "TestWebhook".

image

File: src/pages/Settings.vue [hunk 2]
```diff
@@ -337,6 +360,52 @@ export default {
...
+            let time = dayjs().format("YYYY_MM_DD-hh_mm_ss");
+            let fileName = `Uptime_Kuma_Backup_${time}.json`;
+            let monitorList = Object.values(this.$root.monitorList);
+            let exportData = {
+                version: this.$root.info.version,
+                notificationList: this.$root.notificationList,
+                monitorList: monitorList,
+            }
+            exportData = JSON.stringify(exportData);
+            let downloadItem = document.createElement("a");
```
Comment (louislam) [inline src/pages/Settings.vue:373]: checked, exportData is ok, should be the problem of downloadItem, which remove all spaces unexpectedly.

Comment (Ponkhy) [reply src/pages/Settings.vue:373]: Thank you, should be fixed with [acf6f18](https://github.com/louislam/uptime-kuma/pull/306/commits/acf6f1883b829449cd0883185b301f3e599ecb39)

**Pull request merged**